src/storage/v3/storage.cpp

```diff
@@ -515,38 +491,46 @@ ResultSchema<VertexAccessor> Storage::Accessor::CreateVertexAndValidate(
     return {std::move(*maybe_schema_violation)};
   }
   OOMExceptionEnabler oom_exception;
-  auto gid = storage_->vertex_id_.fetch_add(1, std::memory_order_acq_rel);
+  // Extract key properties
+  std::vector<PropertyValue> primary_properties;
+  for (const auto &[property_id, property_value] : properties) {
+    if (storage_->schemas_.IsPropertyKey(primary_label, property_id)) {
+      primary_properties.push_back(property_value);
+    }
+  }
```

Comment:
What about the order of the properties? I think this way it only works if the primary keys are sorted in the same order in `properties` as the regarding property ids in the schema.